app/src/lib/fetch.ts

```diff
@@ -1,25 +1,27 @@
 import { getAuthHeader } from "./auth/getAuthHeader";
+import { ApiError } from "./errors";
 
 export interface Fetcher {
   url: string;
   method?: "GET" | "POST" | "PUT" | "PATCH" | "DELETE";
   body?: any;
   json?: boolean;
+  noAuth?: boolean;
 }
 
-export const fetcher = async ({ url, method = "GET", json = true, body }: Fetcher) => {
+export const fetcher = async ({ url, method = "GET", json = true, body, noAuth }: Fetcher) => {
   const res = await fetch(url, {
     method,
     ...(body && { body: JSON.stringify(body) }),
     headers: {
       Accept: "application/json",
       "Content-Type": "application/json",
-      ...getAuthHeader()
+      ...(!noAuth && getAuthHeader())
     }
   });
 
   if (!res.ok) {
-    throw new Error("API error");
+    throw new ApiError("API error", res.status);
```

Comment:
There is potentially user-readable information coming back in the API errors, so I'm not sure it should always just hardcode the message to "API error" like this. But maybe not something to address now.

Reply:
yeah currently as issue discuessed above with 401 with no body.. so after that fix can sort